### PR TITLE
Add proxy to log rotation and update syndic rotation for SUSE

### LINKED ISSUE #57353: [BUG] Proxy logs are not rotated
**Description**
Salt Proxy logs are not getting rotated, and need an entry in salt-common.logrotate similar to master and minion

**Setup**
Native minion on Juniper, but applicable for any proxy minion usage.

**Steps to Reproduce the behavior**
Reported by Juniper when developing native minion with make use of proxy

**Expected behavior**
Expect proxy logs not to consume all space on a disk but rotate.

**Screenshots**
From email thread:
```
On QFX10002:
/var/log/salt

% ls -lrt
total 948032
-rw-r-----  1 root  wheel          0 Apr  8 04:55 minion
-rw-r-----  1 root  wheel  485228432 May 17 23:47 proxy

% du -sm proxy
463         proxy
% du -sm /var/log
475         /var/log

% df /var/log
Filesystem  512-blocks   Used  Avail Capacity  Mounted on
/dev/ada1p2     973368 972472 -76968   109%    /.mount/var/log
```

**Versions Report**
```
root@Unknown:~# salt junos174RS8 test.versions
junos174RS8:
    Salt Version:
               Salt: 3000.1-12-gf0937a3fe4
     
    Dependency Versions:
               cffi: 1.14.0
           cherrypy: Not Installed
           dateutil: Not Installed
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
             Jinja2: 2.11.2
            libgit2: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.6.2
       mysql-python: Not Installed
          pycparser: 2.17
           pycrypto: Not Installed
       pycryptodome: 3.9.7
             pygit2: Not Installed
             Python: 3.7.7 (default, Apr 13 2020, 03:51:14)
       python-gnupg: Not Installed
             PyYAML: 5.1.2
              PyZMQ: 19.0.1
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.5.3
                ZMQ: 4.3.1
     
    System Versions:
               dist:   
             locale: utf-8
            machine: amd64
            release: JNPR-11.0-20190825.5ca39af_buil
             system: FreeBSD
            version: Not Installed
```

**Additional context**

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Adds proxy logs to log rotation

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/57353

### Previous Behavior
Proxy logs could fill available disk space

### New Behavior
Proxy logs are now part of log rotation

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dmurphy18** [comment]
Needs 3 reviewers

**Review by s0undt3ch — DISMISSED** [review]

File: pkg/suse/salt-common.logrotate [hunk 1]
```diff
@@ -33,3 +33,19 @@
 	compress
 	notifempty
 }
+
+/var/log/salt/syndic {
```
Comment (DmitryKuzmenko) [inline pkg/suse/salt-common.logrotate:37]: Don't you need `su salt salt` here as above?

Comment (dmurphy18) [reply pkg/suse/salt-common.logrotate:37]: Don't understand the comment 'su salt salt'.
/var/log/salt/syndic is the syndic log in /var/log/salt and already in regular salt, as in, pkg/salt-common.logrotate, and noticed it missing from suse version ().  Adding /var/log/salt/proxy for both regular salt and suse.  The salt-common.logrotate file is used at install time from build packages and is getting installed anyway.

**Review by Akm0d — DISMISSED** [review]

File: pkg/suse/salt-common.logrotate (hunk 1 above)
Comment (DmitryKuzmenko) [reply pkg/suse/salt-common.logrotate:37]: I mean all the records in this logrotate config start either from `su salt salt` or from `su root root`. I don't know why, just found it and pointed your attention to this. Sorry if I'm wrong about that.

Comment (dmurphy18) [reply pkg/suse/salt-common.logrotate:37]: @DmitryKuzmenko Opened up the file and viola, see what you mean, will add the su salt salt, thanks for catching that

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by dmurphy18** [comment]
@dwoz this is a simple fix and needs 3 reviewers, mocking a log rotation does not compute.
Adding previously altered log rotation for syndic to SUSE which was missed at the time.

**Pull request merged**